.github/workflows/exchange-ci.yml

```diff
@@ -0,0 +1,303 @@
+name: Exchange - Continuous Integration
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'exchange/**'
+
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'exchange/**'
+
+  workflow_dispatch:
+
+# We only care about the latest revision of a PR, so cancel all previous instances.
+concurrency:
+  group: ci-${{ github.event.pull_request.number || github.ref_name }}
```

Comment:
this needs to have a unique value for each ci

`ci-exchange-${{ github.event.pull_request.number || github.ref_name }}`